addons/sourcemod/scripting/nd_offmap_fixes.sp

```diff
@@ -31,54 +32,65 @@ int tmpAxisCount;
 int tmpAxisViolated;
 
 //Version is auto-filled by the travis builder
-public Plugin myinfo = 
+public Plugin myinfo =
 {
-	name 		= "[ND] Off Map Buildings Fixes",
-	author 		= "yed_, stickz",
-    	description 	= "Prevents building things in glitched locations",
-    	version 	= "dummy",
-	url 		= "https://github.com/stickz/Redstone/"
+    name        = "[ND] Off Map Buildings Fixes",
+    author      = "yed_, stickz, databomb",
+    description = "Prevents building things in glitched locations",
+    version     = "dummy",
+    url         = "https://github.com/stickz/Redstone/"
+}
+
+// Check for dependency on nd_structure_intercept
+public void OnAllPluginsLoaded()
+{
+    if (!LibraryExists("nd_structure_intercept"))
+    {
+        SetFailState("Failed to find plugin dependency nd_structure_intercept");
```

Comment:
We don't need `SetFailState`. We are hooking a forward, which does not require a dependency. We should check if the library exists instead and use `ND_OnStructureCreated` instead if it doesn't.

Reply:
See https://github.com/stickz/Redstone/commit/398c3cdea39952800872bbc88c20cad3b1dbacc2